src/providers/cohere/chatComplete.ts

```diff
@@ -16,23 +15,39 @@ export const CohereChatCompleteConfig: ProviderConfig = {
     default: 'command',
     required: true,
   },
-  messages: {
-    param: 'prompt',
-    required: true,
-    transform: (params: Params) => {
-      let prompt: string = '';
-      // Transform the chat messages into a simple prompt
-      if (!!params.messages) {
-        let messages: Message[] = params.messages;
-        messages.forEach((msg) => {
-          prompt += `${msg.role}:${msg.content}\n`;
-        });
-        prompt = prompt.trim();
-      }
+  messages: [
+    {
+      param: 'prompt',
+      required: true,
+      transform: (params: Params) => {
+        const messages = params.messages || [];
+        const prompt = messages.at(-1);
+        if (!prompt) {
+          throw new Error('messages length should be at least of length 1');
+        }
 
-      return prompt;
+        return prompt;
+      },
     },
-  },
+    {
+      param: 'chat_history',
+      required: false,
+      transform: (params: Params) => {
+        const messages = params.messages || [];
+        const messagesWithoutLastMessage = messages.slice(
+          0,
+          messages.length - 1
+        );
+        // generate history and forware it to model
+        const history: (Message & { message?: string })[] =
+          messagesWithoutLastMessage.map((message) => ({
+            role: message.role,
+            message: message.content as string,
```

Comment:
also need to handle messages of object type
```
            {
        "role": "assistant",
        "content": [
          {
            "type": "text",
            "text": "is it me?"
          }
        ]
      }
```
You can refer https://platform.openai.com/docs/api-reference/chat/create for openai request structure.
Gateway transforms requests from OpenAI format to provider specific format, so all the possible request variations from. openai need to be handled by the gateway transformer